### PR TITLE
Comment cleanup and rename of DsState::Repair -> Reconcile

### PR DESCRIPTION
**Pull request by leftwo**
Some misc comment cleanup and a rename of the Downstairs state Repair to Reconcile.  The term "repair" was often used in both the reconcile case (what happens at start up) and during live repair (what happens when the upstairs is active).  This cleans up the comments and renames the original DsState::Repair to be called DsState::Reconcile.

There should be no functional change from this.

### CODE REVIEW

File: openapi/crucible-control.json [hunk 1]
```diff
@@ -96,8 +96,8 @@
           "wait_quorum",
           "bad_region",
           "disconnected",
-          "repair",
-          "failed_repair",
+          "reconcile",
+          "failed_reconcile",
```
Comment (mkeeter) [inline openapi/crucible-control.json:100]: Is anyone using these APIs that would need to change?

File: upstairs/src/downstairs.rs [hunk 2]
```diff
@@ -3548,7 +3548,8 @@ impl Downstairs {
     }
 
     #[cfg(test)]
-    /// Create a test downstiars which has one downstairs client transitioned to LiveRepair
+    /// Create a test downstiars which has one downstairs client transitioned to
```
Comment (mkeeter) [inline upstairs/src/downstairs.rs:3551]:
```suggestion
    /// Create a test downstairs which has one downstairs client transitioned to
```

Comment (mkeeter) [reply upstairs/src/downstairs.rs:3551]: (not a new typo, but I saw it while reviewing)

**Review by mkeeter — APPROVED** [review]
This is a much-desired cleanup!

**Review by jmpesp — APPROVED** [review]

File: upstairs/src/downstairs.rs (hunk 2 above)
Comment (leftwo) [reply upstairs/src/downstairs.rs]: I went and found and replaced all the downstiars cases.

**Pull request merged**